### PR TITLE
Extract bundle fluentd plugins function

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Testing performed

- [x] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: ci/build.sh [hunk 1]
```diff
@@ -18,6 +18,39 @@ function get_branch_to_checkout() {
     || echo "${TRAVIS_BRANCH}"
 }
 
+function bundle_fluentd_plugins() {
+  local VERSION="${1}"
+
+  if [[ "${VERSION}" == "" ]] ; then
+    echo "Please provide the version when bundling fluentd plugins"
+    exit 1
+  fi
```
Comment (sumo-drosiek) [inline ci/build.sh:27]: Does this check do anything? Is it possible to not pass VERSION as we set it up in header?

Comment (pmalek-sumo) [reply ci/build.sh:27]: With the current `VERSION` variable declaration it doesn't really do anything. Better safe than sorry. I can remove it if don't want this here

Comment (sumo-drosiek) [reply ci/build.sh:27]: Just wanted to ensure

File: ci/build.sh [hunk 2]
```diff
@@ -18,6 +18,39 @@ function get_branch_to_checkout() {
     || echo "${TRAVIS_BRANCH}"
 }
 
+function bundle_fluentd_plugins() {
+  local VERSION="${1}"
```
Comment (perk-sumo) [inline ci/build.sh:22]: 😍

File: ci/build.sh [hunk 3]
```diff
@@ -18,6 +18,39 @@ function get_branch_to_checkout() {
...
+
+  if [[ "${VERSION}" == "" ]] ; then
```
Comment (perk-sumo) [inline ci/build.sh:24]: One could also use `local VERSION="${1:?}"` above to check if the variable is set but I like the explicit error in here 👍

File: ci/build.sh [hunk 4]
```diff
@@ -18,6 +18,39 @@ function get_branch_to_checkout() {
...
+    echo "Please provide the version when bundling fluentd plugins"
+    exit 1
+  fi
+
+  for i in ./fluent-plugin-*/ ; do
+    if [[ -d "${i}" ]]; then
+    (
+      cd "${i}" || exit 1
+      PLUGIN_NAME="$(basename "${i}")"
```
Comment (perk-sumo) [inline ci/build.sh:33]:
```suggestion
      local plugin_name="$(basename "${i}")"
```

File: ci/build.sh [hunk 5]
```diff
@@ -18,6 +18,39 @@ function get_branch_to_checkout() {
...
+      # Strip everything after "-" (longest match) to avoid gem prerelease behavior
+      GEM_VERSION="${VERSION%%-*}"
```
Comment (perk-sumo) [inline ci/build.sh:35]:
```suggestion
      local gem_version="${version%%-*}"
```

File: ci/build.sh (hunk 2 above)
Comment (perk-sumo) [reply ci/build.sh:22]: Let's change the VERSION variable into `version` to indicate it's a local one.
```suggestion
  local version="${1}"
```

**Review by astencel-sumo — APPROVED** [review]

File: ci/build.sh (hunk 3 above)
Comment (perk-sumo) [inline ci/build.sh:24]:
```suggestion
  if [[ "${version}" == "" ]] ; then
```

File: ci/build.sh [hunk 6]
```diff
@@ -18,6 +18,39 @@ function get_branch_to_checkout() {
...
+      echo "Building gem ${PLUGIN_NAME} version ${GEM_VERSION} in $(pwd) ..."
+      sed -i.bak "s/0.0.0/${GEM_VERSION}/g" ./"${PLUGIN_NAME}".gemspec
+      rm -f ./"${PLUGIN_NAME}".gemspec.bak
```
Comment (perk-sumo) [inline ci/build.sh:38]:
```suggestion
      echo "Building gem ${plugin_name} version ${gem_version} in $(pwd) ..."
      sed -i.bak "s/0.0.0/${gem_version}/g" ./"${plugin_name}".gemspec
      rm -f ./"${plugin_Name}".gemspec.bak
```

**Review by perk-sumo — COMMENTED** [review]
Couple of nitpicks. Let me create another PR with them.

File: ci/build.sh (hunk 1 above)
Comment (perk-sumo) [reply ci/build.sh:27]: It's good to know if version is there.

**Pull request merged**